### ISSUE TITLE
Should we store population data or a `symmetric` boolean in `attr`?

### ISSUE THREAD

**Issue by Bisaloo** [socialcontactdata/contactmatrix#9]
We have two options in `attr`:

- a population data vector
- a boolean indicating if the matrix is symmetric

We can get the latter from the former. But the former is also not always available in case where the latter would be.

_Originally posted by @sbfnk in https://github.com/socialcontactdata/contactmatrix/pull/6#discussion_r1584756433_

**Comment by sbfnk** [comment]
I think even if the latter is available the matrix will only ever be symmetric with respect to a specific population. That's why I think the early return in https://github.com/socialcontactdata/contactmatrix/blob/40c12976e061942fe9005b2bd4bef852ca11d2ed/R/cm_make_symmetric.R#L20 is wrong and we shouldn't store either of the two properties but require population data as an arguments for specific operations such as making the matrix symmetric.

**Comment by Bisaloo** [comment]
Yes, it makes sense. This entirely defeats the purpose of the `symmetric` boolean attribute then. Let's go with storing the population data vector.

**Comment by sbfnk** [comment]
Frankly I'm not sure we should store either (and I'm leaning towards removing the population data from the return of `contact_matrix()`) - given a matrix the population vector could be:
1) the sample population (say, participants in a survey)
2) the population sampled from (say, the UK population)
3) the target population, i.e. some other population that I'm applying this to in my model (say, the population of a country I'm modelling and for which I don't have data)

It might be clearer to always require a population argument explicitly for operations that need it (e.g. making counts symmetric).
